airflow/models/abstractoperator.py

```diff
@@ -395,6 +395,18 @@ def get_closest_mapped_task_group(self) -> MappedTaskGroup | None:
         """
         return next(self.iter_mapped_task_groups(), None)
 
+    def get_needs_expansion(self):
+        """
+        Returns true of the task is mapped or is in a mapped task group.
+
+        :meta private:
+        """
+        if self._needs_expansion is not None:
+            return self._needs_expansion
+        if self.get_closest_mapped_task_group() is not None:
+            return True
+        return False
```

Comment:
sure we can store the value.

re property... it can be ... do you think it should be?  then the suggestion to make it a cached property will come. but then it's unclear how we persist the cache on serialization round trip with cached prop?

Reply:
i pushed changes caching the value.  please look and lemme know.

Reply:
I don’t have a preference either way tbh. An argument-less method and a property are basically the same in Python anyway (aside from the call-site syntax). Using `cache_property` is not _technically_ out of the question since we need to manually populate the underlying value—we _can_ do that with a cached property too, it’s just very awkward. I think anyone trying to raise that suggestion would quickly figure that out when they try to actually change the implementation anyway.

Reply:
yeah i don't really care either way, and since it's private it doesn't really matter, and there are not strong feelings on either side, so i'll leave it now